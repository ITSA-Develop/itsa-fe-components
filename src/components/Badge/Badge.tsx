import { Badge as AntBadge, BadgeProps } from 'antd';

export const Badge = ({ ...rest }: BadgeProps) => {
	return <AntBadge {...rest} />;
};
