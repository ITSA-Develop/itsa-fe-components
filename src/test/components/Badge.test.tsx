import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { Badge } from '../../components/Badge/Badge';

describe('Badge component', () => {
	it('renders children and shows badge count', () => {
		const { container } = render(
			<Badge count={5}>
				<span data-testid="child">Inbox</span>
			</Badge>,
		);

		expect(screen.getByTestId('child')).toBeInTheDocument();
		expect(screen.getByText('5')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('renders children without count', () => {
		const { container } = render(
			<Badge>
				<span data-testid="child">No count</span>
			</Badge>,
		);

		expect(screen.getByTestId('child')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('shows dot badge when dot prop is true', () => {
		const { container } = render(
			<Badge dot>
				<span data-testid="child">Dot badge</span>
			</Badge>,
		);

		expect(document.querySelector('.ant-badge-dot')).toBeTruthy();
		expect(container).toMatchSnapshot();
	});
});
