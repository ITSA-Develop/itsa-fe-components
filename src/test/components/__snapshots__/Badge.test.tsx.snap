// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge component > renders children and shows badge count 1`] = `
<div>
  <span
    class="ant-badge css-dev-only-do-not-override-l9pxc0"
  >
    <span
      data-testid="child"
    >
      Inbox
    </span>
    <sup
      class="ant-scroll-number ant-badge-count"
      data-show="true"
      title="5"
    >
      <bdi>
        <span
          class="ant-scroll-number-only"
          style="transition: none;"
        >
          <span
            class="ant-scroll-number-only-unit current"
          >
            5
          </span>
        </span>
      </bdi>
    </sup>
  </span>
</div>
`;

exports[`Badge component > renders children without count 1`] = `
<div>
  <span
    class="ant-badge css-dev-only-do-not-override-l9pxc0"
  >
    <span
      data-testid="child"
    >
      No count
    </span>
  </span>
</div>
`;

exports[`Badge component > shows dot badge when dot prop is true 1`] = `
<div>
  <span
    class="ant-badge css-dev-only-do-not-override-l9pxc0"
  >
    <span
      data-testid="child"
    >
      Dot badge
    </span>
    <sup
      class="ant-scroll-number ant-badge-dot"
      data-show="true"
    />
  </span>
</div>
`;
